orchagent/pfcwdorch.cpp

```diff
@@ -42,6 +42,45 @@ PfcWdOrch<DropHandler, ForwardHandler>::PfcWdOrch(DBConnector *db, vector<string
     timer->start();
 }
 
+template <typename DropHandler, typename ForwardHandler>
+QueueMcDropCounters PfcWdOrch<DropHandler, ForwardHandler>::getQueueMcDropCounters(
+        const Port& port,
+        uint8_t pfcMask)
+{
+    SWSS_LOG_ENTER();
+
+    vector<FieldValueTuple> fieldValues;
+    QueueMcDropCounters counters;
+    RedisClient redisClient(PfcWdOrch<DropHandler, ForwardHandler>::getCountersDb().get());
+
+    for (uint8_t prio = 0; prio < port.m_queue_ids.size(); prio++)
+    {
+        sai_object_id_t queueId = port.m_queue_ids[prio + PFC_WD_TC_MAX];
+        auto queueIdStr = sai_serialize_object_id(queueId);
+        auto queueType = redisClient.hget(COUNTERS_QUEUE_TYPE_MAP, queueIdStr);
+
+        if (queueType.get() == nullptr || *queueType != "SAI_QUEUE_TYPE_MULTICAST" || !m_countersTable->get(queueIdStr, fieldValues))
+        {
+            continue;
+        }
+
+        uint64_t drops = numeric_limits<uint64_t>::max();
+        for (const auto& fv : fieldValues)
+        {
+            const auto field = fvField(fv);
+            const auto value = fvValue(fv);
+
+            if (field == "SAI_QUEUE_STAT_PACKETS")
```

Comment:
>SAI_QUEUE_STAT_PACKETS [](start = 26, length = 22)

this is tx counters, why later treat this as drop counters?

Reply:
I wasn't a part of the original discussions Guohan, Matty and Liat had in December re dropping multicast traffic on all loseless traffic.
However, my understanding in the internal discussions I had afterwards with Matty, was that since we need to drop all the traffic in the loseless multicast queues, we want to show the amount of dropped multicast packets under the queue stat packets, for multicast queues, because every packet is dropped, so total = dropped
If you think it's confusing, I suggest what we can do in that case, is to show this information for multicast queues, as SAI_QUEUE_STAT_DROPPED_PACKETS, instead of SAI_QUEUE_STAT_PACKETS

Reply:
This was a mistake from my side
It's a tx counters, I shouldn't have named them "Dropped packets"